configure

```diff
@@ -732,13 +732,18 @@ else
   echo "Checking for strerror... No." | tee -a configure.log
 fi
 
-# check for getauxval() for architecture feature detection at run-time
+# check for getauxval() or elf_aux_info() for architecture feature detection at run-time
 cat > $test.c <<EOF
 #include <sys/auxv.h>
+#ifdef __FreeBSD__
+#include <machine/cpu.h>
+int main() { unsigned long hwcap; return elf_aux_info(AT_HWCAP, &hwcap, sizeof(hwcap)); }
+#else
 int main() { return getauxval(0); }
+#endif
 EOF
 if try $CC $CFLAGS -o $test $test.c $LDSHAREDLIBC; then
-  echo "Checking for getauxval() in sys/auxv.h... Yes." | tee -a configure.log
+  echo "Checking for getauxval() or elf_aux_info() in sys/auxv.h... Yes." | tee -a configure.log
```

Comment:
Now `in sys/auxv.h` would be misleading...

Reply:
Why is it misleading? `elf_aux_info` is literally in `sys/auxv.h`.

Reply:
Because for other platforms it is supposed to test if `sys/auxv.h` includes all necessary headers automatically, which isn't always true even for Linux.

Reply:
Please check now, it doesn't require any headers other than `sys/auxv.h` anymore.

Reply:
I will check next time I boot FreeBSD 13... We still need @Dead2 to enable the CI tests so we know this PR doesn't break build on other platforms.